src/io/github/sspanak/tt9/ime/TraditionalT9.java

```diff
@@ -349,19 +349,20 @@ public boolean onKeyAddWord() {
 
 	public boolean onKeyNextLanguage() {
 		if (nextLang()) {
-			commitCurrentSuggestion(false);
 			mInputMode.changeLanguage(mLanguage);
-			mInputMode.reset();
-			resetKeyRepeat();
```

Comment:
This is still causing problems. Try changing to 123 mode, pressing any number, then changing the language. It may cause other side effects, so key repeat must be reset at some point, but without breaking ABC mode.